### PR TITLE
check chdir/mkdir return codes

### PR DESCRIPTION
**Pull request by VisualPrankDude**
no text

### CODE REVIEW

File: binr/r2pm/r2pm [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 # r2pm
-
+chdir() { cd "${1}" || exit 1; }
+mkdir() { mkdir "${@}" || exit 1; }
```
Comment (radare) [inline binr/r2pm/r2pm]: thats wrong

File: binr/r2pm/r2pm [hunk 2]
```diff
@@ -233,13 +234,13 @@ R2PM_SVN() {
 	else
 		DIR="$2"
 	fi
-	cd "${R2PM_GITDIR}" || exit 1
+	chdir "${R2PM_GITDIR}" || exit 1
```
Comment (radare) [inline binr/r2pm/r2pm]: this one checks for errors already

File: binr/r2pm/r2pm [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 # r2pm
-
+chdir() { cd "${1}" || exit 1; }
```
Comment (radare) [inline binr/r2pm/r2pm]: dont add macros when we just need to catch the || R2PM_FAIL in each case.

**Comment by radare** [comment]
Ping pls update

**Comment by VisualPrankDude** [comment]
I was under the impression, we use "test -d" and checked for non-readable/executable directories. See difference:

,mkdir -m 0000 rpmdir
,cd rpmdir || echo fail
bash: cd: rpmdir: Permission denied
fail
,pwd
/tmp
,test -d rpmdir || echo fail

**Comment by VisualPrankDude** [comment]
@radare see above

**Comment by VisualPrankDude** [comment]
@radare this was the exact lines, I stumbled over:

+       if [ -d "${DIR}" ]; then
+               cd "${DIR}"
+               svn up

**Comment by radare** [comment]
Yep i agree in that but i dont want the functions i think its better to handle errors on each use and not create those functions

**Comment by VisualPrankDude** [comment]
colsing

**Pull request closed without merging**